guillotina/contrib/memcached/driver.py

```diff
@@ -0,0 +1,110 @@
+try:
+    import emcache
+    import emcache.client_errors
+except ImportError:
+    print("If you add guillotina.contrib.memcached you need to add pfreixes/emcache on your requirements")
+    raise
+
+from guillotina import app_settings
+from guillotina.contrib.memcached.exceptions import NoMemcachedConfigured
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+import asyncio
+import backoff
+import logging
+
+
+logger = logging.getLogger("guillotina.contrib.memcached")
+
+
+class MemcachedDriver:
+    """
+    Implements a cache driver using Memcached
+    """
+
+    def __init__(self):
+        self._client: Optional[emcache.Client] = None
+        self.initialized: bool = False
+        self.init_lock = asyncio.Lock()
+
+    @property
+    def client(self):
+        return self._client
+
+    async def initialize(self, loop):
+        async with self.init_lock:
+            if self.initialized is False:
+                while True:
+                    try:
+                        await self._connect()
+                        self.initialized = True
+                        break
+                    except Exception:  # pragma: no cover
+                        logger.error("Error initializing pubsub", exc_info=True)
+
+    @backoff.on_exception(backoff.expo, (OSError,), max_time=30, max_tries=4)
+    async def _connect(self):
+        settings = app_settings["memcached"]
+        hosts = [emcache.MemcachedHostAddress(host, int(port)) for host, port in settings["hosts"]]
+        if len(hosts) == 0:
+            raise NoMemcachedConfigured()
+        kw = {}
+        for key in ("timeout", "max_connections"):
```

Comment:
why not support all constructor parameters from day zero [1]? They are just a handful but some of them might become really useful. 

For example `purge_unhealthy_nodes` allows you to remove temporary nodes from the pool - with a cascade of misses as a side effect - which they are considered unhealthy.


[1] https://emcache.readthedocs.io/en/latest/client.html

Reply:
Also `connection_timeout ` is an important one.

Reply:
done all params except `cluster_events`